### ISSUE TITLE
Feature: Print the name of the file being linted

### ISSUE THREAD

**Issue by mcandre** [rodjek/puppet-lint#283]
If a Puppet file has lint warnings, please preface the list of warnings with the name of the file. When puppet-lint'ing a large directory, showing the filename helps determine which files are triggering the warnings.

Currently, puppet-lint does not do this, so running:

```
find . -type f -name '*.pp' -exec puppet-lint {} \;
```

may produce hundreds of warnings, but never saying which files trigger them.

As a workaround, one can type:

```
find . -type f -name '*.pp' -print -exec puppet-lint {} \;
```

But this prints out all names, not just those that have warnings.
